inc/domainrecord.class.php

```diff
@@ -208,6 +222,18 @@ private function prepareInput($input) {
          $input['Status'] = self::STATUS_ACTIVE;
       }
 
+      if (isset($input['domainrecordtypes_id'])) {
+         if (!($_SESSION['glpiactiveprofile']['managed_domainrecordtypes'] == [-1]
+            ||in_array($this->fields['domainrecordtypes_id'], $_SESSION['glpiactiveprofile']['managed_domainrecordtypes']))) {
```

Comment:
On creation, you should check `$input['domainrecordtypes_id']`, and on update, you should check input and previous value.
Error message should also be adapted.

Reply:
Done